server/routerlicious/packages/lambdas/src/scribe/pendingMessageReader.ts

```diff
@@ -0,0 +1,23 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ISequencedDocumentMessage } from "@fluidframework/protocol-definitions";
+import { IDeltaService } from "@fluidframework/server-services-core";
+import { getLumberBaseProperties, Lumberjack } from "@fluidframework/server-services-telemetry";
+import { IPendingMessageReader } from "./interfaces";
+
+export class PendingMessageReader implements IPendingMessageReader {
+    constructor(
+        private readonly tenantId: string,
+        private readonly documentId: string,
+        protected readonly deltaService: IDeltaService,
+    ) { }
+    
+    public async readMessages(from: number, to: number): Promise<ISequencedDocumentMessage[]> {
+        Lumberjack.info(`Fetching pending messages from ${from} to ${to}`, getLumberBaseProperties(this.documentId, this.tenantId));
+        const deltasP = this.deltaService.getDeltas("", this.tenantId, this.documentId, from-1, to+1);
```

Comment:
Is this empty string `""` parameter the collection name? Is it ok for this to be empty?

Reply:
I realized I got confused and this deltaService is not the one defined here: https://github.com/microsoft/FluidFramework/blob/15c922f721519ab72c93a787e54738b8e76ccf74/server/routerlicious/packages/routerlicious-base/src/alfred/services/deltaService.ts#L21

But instead, the one defined below in this PR. To make sure I understand it correctly: is the reason for creating this new IDeltaService implementation that the original would directly query the DB for the deltas, wheres instead, here in Scribe, we want to instead exercise the Alfred get deltas API?

If that is the case, should we enforce that this `PendingMessageReader` take not a generic `IDeltaService` (which could be the one that only talks to the DB), but instead, the implementation that calls the Alfred APIs? Or maybe this was intentional if somebody using r11s doesn't want to have the request go through Alfred?

Reply:
Correct on both! I explained the first point in the description of the PR. As to the second point, we only need people to adhere to the contract. They can get the deltas however they want.

Reply:
Got it! Thanks for the info. I did not realize that was the case when I read the description